File: VB/SnxToRepx/Converter/ValueComparers.vb
Imports DevExpress.XtraRichEdit.API.Layout
Imports System.Collections.Generic

Namespace SnxToRepx.Converter

    ''' <summary>
    ''' This is a comparer to sort a collection of RangedLayoutElements
    ''' </summary>
    Friend Class RangedElementComparer
        Implements IComparer(Of RangedLayoutElement)

        ''' <summary>
        ''' Compares two RangedLayoutElements by their ranges
        ''' </summary>
        ''' <param name="x">The first element</param>
        ''' <param name="y">The second element</param>
        ''' <returns>The result of comparison</returns>
        Public Function Compare(ByVal x As RangedLayoutElement, ByVal y As RangedLayoutElement) As Integer Implements IComparer(Of RangedLayoutElement).Compare
            'Use the default integer comparer to compare start positions for elements
            Return Comparer(Of Integer).Default.Compare(x.Range.Start, y.Range.Start)
        End Function
    End Class
End Namespace

File: CS/SnxToRepx/Converter/ValueComparers.cs
using DevExpress.XtraRichEdit.API.Layout;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SnxToRepx.Converter
{
    /// <summary>
    /// This is a comparer to sort a collection of RangedLayoutElements
    /// </summary>
    class RangedElementComparer : IComparer<RangedLayoutElement>
    {
        /// <summary>
        /// Compares two RangedLayoutElements by their ranges
        /// </summary>
        /// <param name="x">The first element</param>
        /// <param name="y">The second element</param>
        /// <returns>The result of comparison</returns>
        public int Compare(RangedLayoutElement x, RangedLayoutElement y)
        {
            //Use the default integer comparer to compare start positions for elements
            return Comparer<int>.Default.Compare(x.Range.Start, y.Range.Start);
        }
    }
}
